박정근/Programmers/Week39/야근_지수.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Author    : Park Jeong Geun
+ * Date      : 2025.03.23(Sun)
+ * Runtime   : 1.91ms
+ * Memory    : 91.1MB
+ * Algorithm : parametric-search
+ */
+
+// >> 첫 번째 풀이 ( parametric-search )
+// 파라매트릭 서치로 풀이했습니다. (Solve Time : 0h 45m)
+
+// n만큼 works 배열을 줄이되, 원소마다 차이를 최대한으로 줄여야 하는 문제입니다.
+// 파라매트릭 서치로, 작업했을 때 works 배열의 최대 잔여량을 구합니다. 
+// n 보다 더 작업량이 나온다면, 최대 잔여량만큼 남은 원소들을 1씩 더하면서 나머지들과 같이 제곱 합해주면 답이 됩니다.
+
+class Solution {
+    public long solution(int n, int[] works) {
+        long answer = 0;
+        
+        // 1. 파라매트릭 서치로, 작업했을 때 works 배열의 최대 잔여량을 구합니다.
+        int start = 0;
+        int end = max_works;
```

Comment:
최대 잔여량 초기화가 잘못 되있는 것 같습니다..!

```suggestion
        assert works.length > 0;
        int end = Arrays.stream(works).max().getAsInt();
```

Reply:
초반 코드가 일부 날라간 채로 커밋되었네요. 수정해서 다시 업로드하겠습니다.

Reply:
수정했습니다. 20~25 행이 추가되었습니다. (sum_works, max_works)